### PR TITLE
[perf] Load localizations lazily

### PR DESCRIPTION
**Pull request by msujew**
#### What it does

Related to https://github.com/eclipse-theia/theia/issues/12924

Makes the localization entries used in the `LocalizationProvider` lazy. This is mainly relevant for localizations contributed by language packs, as they are usually split into dozens of files, which takes quite a while to load and parse. Each language pack takes roughly ~50-100ms to load. Having 10+ language packs installed can slow down startup time considerably.

#### How to test

1. Load a language pack and assert that it still works as expected.
2. Install all language packs (see [here](https://open-vsx.org/?category=Language%20Packs)) and assert that the startup time isn't influenced (at least not noticably).

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by msujew** [comment]
cc @kittaakos This should help improve the startup performance in the Arduino IDE. I'd appreciate some feedback :)

**Comment by kittaakos** [comment]
I checked out the branch and made a quick comparison.

My changeset:

`package.json`:
```diff
diff --git a/package.json b/package.json
index 929fb05554..32b8de6e19 100644
--- a/package.json
+++ b/package.json
@@ -100,11 +100,23 @@
   ],
   "theiaPluginsDir": "plugins",
   "theiaPlugins": {
-    "eclipse-theia.builtin-extension-pack": "https://open-vsx.org/api/eclipse-theia/builtin-extension-pack/1.79.0/file/eclipse-theia.builtin-extension-pack-1.79.0.vsix",
-    "EditorConfig.EditorConfig": "https://open-vsx.org/api/EditorConfig/EditorConfig/0.16.6/file/EditorConfig.EditorConfig-0.16.6.vsix",
-    "dbaeumer.vscode-eslint": "https://open-vsx.org/api/dbaeumer/vscode-eslint/2.4.2/file/dbaeumer.vscode-eslint-2.4.2.vsix",
-    "ms-vscode.js-debug": "https://open-vsx.org/api/ms-vscode/js-debug/1.78.0/file/ms-vscode.js-debug-1.78.0.vsix",
-    "ms-vscode.js-debug-companion": "https://open-vsx.org/api/ms-vscode/js-debug-companion/1.1.2/file/ms-vscode.js-debug-companion-1.1.2.vsix"
+    "vscode-language-pack-bg": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-bg/1.48.3/file/MS-CEINTL.vscode-language-pack-bg-1.48.3.vsix",
+    "vscode-language-pack-cs": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-cs/1.78.0/file/MS-CEINTL.vscode-language-pack-cs-1.78.0.vsix",
+    "vscode-language-pack-de": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-de/1.78.0/file/MS-CEINTL.vscode-language-pack-de-1.78.0.vsix",
+    "vscode-language-pack-es": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-es/1.78.0/file/MS-CEINTL.vscode-language-pack-es-1.78.0.vsix",
+    "vscode-language-pack-fr": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-fr/1.78.0/file/MS-CEINTL.vscode-language-pack-fr-1.78.0.vsix",
+    "vscode-language-pack-hu": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-hu/1.48.3/file/MS-CEINTL.vscode-language-pack-hu-1.48.3.vsix",
+    "vscode-language-pack-it": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-it/1.78.0/file/MS-CEINTL.vscode-language-pack-it-1.78.0.vsix",
+    "vscode-language-pack-ja": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-ja/1.78.0/file/MS-CEINTL.vscode-language-pack-ja-1.78.0.vsix",
+    "vscode-language-pack-ko": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-ko/1.78.0/file/MS-CEINTL.vscode-language-pack-ko-1.78.0.vsix",
+    "vscode-language-pack-nl": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-nl/1.48.3/file/MS-CEINTL.vscode-language-pack-nl-1.48.3.vsix",
+    "vscode-language-pack-pl": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-pl/1.78.0/file/MS-CEINTL.vscode-language-pack-pl-1.78.0.vsix",
+    "vscode-language-pack-pt-BR": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-pt-BR/1.78.0/file/MS-CEINTL.vscode-language-pack-pt-BR-1.78.0.vsix",
+    "vscode-language-pack-ru": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-ru/1.78.0/file/MS-CEINTL.vscode-language-pack-ru-1.78.0.vsix",
+    "vscode-language-pack-tr": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-tr/1.78.0/file/MS-CEINTL.vscode-language-pack-tr-1.78.0.vsix",
+    "vscode-language-pack-uk": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-uk/1.48.3/file/MS-CEINTL.vscode-language-pack-uk-1.48.3.vsix",
+    "vscode-language-pack-zh-hans": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-zh-hans/1.78.0/file/MS-CEINTL.vscode-language-pack-zh-hans-1.78.0.vsix",
+    "vscode-language-pack-zh-hant": "https://open-vsx.org/api/MS-CEINTL/vscode-language-pack-zh-hant/1.78.0/file/MS-CEINTL.vscode-language-pack-zh-hant-1.78.0.vsix"
   },
   "theiaPluginsExcludeIds": [
     "ms-vscode.js-debug-companion",
```

`.vscode/launch.json`:
```diff
diff --git a/.vscode/launch.json b/.vscode/launch.json
index 00e3862ace..37276a42ef 100644
--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -30,7 +30,6 @@
         ".",
         "--log-level=debug",
         "--hostname=localhost",
-        "--no-cluster",
         "--app-project-path=${workspaceFolder}/examples/electron",
         "--remote-debugging-port=9222",
         "--no-app-auto-install",
```

`HEAD` ([f038673](https://github.com/eclipse-theia/theia/commit/f038673dc228485b34e8762686525d8e5f258afb)) of the `main` branch:

https://github.com/eclipse-theia/theia/assets/1405703/d202fb56-2efa-403f-b608-5abdf600e184

From this PR:

https://github.com/eclipse-theia/theia/assets/1405703/8c03169d-bb54-43f6-b450-547372b60caf



I did the profiling with the built-in profiler in VS Code on macOS 13.5.2, 2,6 GHz 6-Core Intel Core i7, 16 GB, starting the electron example app from the sources in cluster mode. Both profiles show the first second in the backend process.

From this PR, I can select and change the language. The app reloads as expected. Excellent work, Mark

File: packages/plugin-ext/src/hosted/node/hosted-plugin-localization-service.ts
Comment (kittaakos) [inline packages/plugin-ext/src/hosted/node/hosted-plugin-localization-service.ts:102]: Could this be improved by collecting all `getL10nTranslations` as promises and running them together?

If there is no dependency between `plugin.contributes?.localizations` and `plugin.metadata.model.l10n` async tasks, maybe running them all together could be an additional improvement. Assuming they're IO (file read)

**Comment by JonasHelming** [comment]
@planger FYI

**Comment by JonasHelming** [comment]
@msujew Do you want to trigger a re-review from @kittaakos ?

**Comment by msujew** [comment]
@JonasHelming I don't need a second review from @kittaakos, nothing has really changed. I just need an approval to merge this. Let me quickly rebase the PR :)

**Review by JonasHelming — APPROVED** [review]

**Pull request merged**
